### PR TITLE
feat: Add {% render %} tag

### LINKED ISSUE #37: Allow passing of parameters to include
Newer versions of liquid allow the `include` tag to pass values. Current workaround is:

```
{% assign snippet_variable = 'this is it' %}
{% assign snippet_variable_two = 'this is also it' %}
{% include 'snippet' %}
```

### PR DESCRIPTION
**Pull request by JasonEtco**
This PR will add the `{% render %}` tag, [according to Shopify's Ruby Liquid docs](https://shopify.dev/docs/themes/liquid/reference/tags/theme-tags#render). This is a tag similar to `include`, but it takes variables and "componentizes" the snippet (the external context can't be used inside a `render`-ed snippet).

### Usage

```
Include (but no shared context):
{% render "file" %}

Render a variable file:
{% render variable_file %}

Pass a variable:
{% render "file", name: "Jason" %}

Pass a variable from the context:
{% render "file", user: user %}

Pass a whole context using `with`:
{% render "file" with user as "Jason" %}

Iterate over a value using `for`:
{% render "file" for users as user %}
```

### Todo

- [x] Write some tests
- [x] Pass those tests (make the actual tag)
- [x] ~Update docs (we don't really have docs on particular tags 🤔)~ I don't think there's anything to document, the official Liquid docs say it all

Closes #37

### CODE REVIEW

**Review by zeke — APPROVED** [review]
Implementation looks good! Easy to follow.

I think it'd be great to include this in the README. Maybe just the contents of this PR body as a start?

File: lib/liquid/tags/render.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+    const src = await this.template
+      .engine
+      .fileSystem
+      .readTemplateFile(this.filepath)
+
+    return this.template.engine.parse(src)
+  }
+
+  async renderAll (subTemplate, variables) {
```
Comment (JasonEtco) [inline lib/liquid/tags/render.js:74]: I think ideally we would re-use `Liquid.For` logic, but it's unfortunately a little complicated. I don't feel like we need it FWIW.

**Comment by JasonEtco** [comment]
> I think it'd be great to include this in the README. Maybe just the contents of this PR body as a start?

@zeke I gave this some real thought and attempts, but I don't think there's a good way to do this. Right now we simply rely on Ruby/Shopify Liquid's docs for the Liquid syntax - we don't have any docs in this repo.

We can add a section to the README like **Built-in tags**, but we'd just be duplicating what's in the official docs. And, I think documenting this one tag would be counterproductive - if all tags aren't listed, it might be more confusing than none.

**Comment by zeke** [comment]
Works for me. 🚢

**Pull request merged**